src/cli/Utils.cpp

```diff
@@ -108,6 +109,22 @@ namespace Utils
         TextStream out(outputDescriptor);
         TextStream err(errorDescriptor);
 
+        QFileInfo dbFileInfo(databaseFilename);
+        if (dbFileInfo.canonicalFilePath().isEmpty()) {
+            out << QObject::tr("Failed to open database file %1: not found").arg(databaseFilename) << endl;
+            return {};
+        }
```

Comment:
@sjamesr could we add unit tests on some of those new checks?

Reply:
Done, there should be coverage for all the checks now.

Reply:
Tests are failing on Windows now
```
FAIL!  : TestCli::testInvalidDbFiles() Compared values are not the same
    Actual   (QString(m_stderrFile->readAll()))                                             : "Error while reading the database: Failed to read database file.\n"
    Expected (QObject::tr("Failed to open database file %1: not readable").arg(path) + "\n"): "Failed to open database file C:/msys64/tmp/testcli.FULFwe: not readable\n"
```

Reply:
I couldn't get isReadable to return false on Windows, even with NTFS permissions enabled. I left the check in but disabled the test on Windows.